code/src/reducers/nutrition.js

```diff
@@ -0,0 +1,118 @@
+import { createSlice } from '@reduxjs/toolkit'
+import fruitVeg from '../assets/fruitVeg.png'
+import dairy from '../assets/dairy.png'
+import freezer from '../assets/freezer.png'
+import meat from '../assets/meat.png'
+
+const initialState = {
+  list: {
+    shelves: [
+      {
+        name: 'Fruit & Veg',
+        reveal: false,
+        products: [],
+        image: fruitVeg
+      },
+      {
+        name: 'Dairy',
+        reveal: false,
+        products: [],
+        image: dairy
+      },
+      {
+        name: 'Meat & Fish',
+        reveal: false,
+        products: [],
+        image: meat
+      },
+      {
+        name: 'Freezer',
+        reveal: false,
+        products: [],
+        image: freezer
+      }
+    ]
+  },
+  productDetails: {
+
+  }
+}
+
+export const nutrition = createSlice({
+  name: 'nutrition',
+  initialState: initialState,
+  reducers: {
+    addProduct: (state, action) => {
+      const { newProduct, shelf } = action.payload
+      newProduct.quantity = 1;
+      const foundShelf = state.list.shelves.find((x) => x.name === shelf)
```

Comment:
To make more readable, consider
`const foundShelf = findShelfByName(shelf)`

Reply:
I understand that it's one line of code and seems like overkill to create a function for this, however, readability _is_ that important 😃